tests/unittests/test_datasource/test_ec2.py

```diff
@@ -692,39 +822,59 @@ def test_convert_ec2_metadata_network_config_handles_local_v4_and_v6(self):
         nic1_metadata['ipv6s'] = '2620:0:1009:fd00:e442:c88d:c04d:dc85/64'
         nic1_metadata.pop('public-ipv4s')
         nic1_metadata['local-ipv4s'] = '10.0.0.42'  # Local ipv4 only on vpc
-        expected = {'version': 1, 'config': [
-            {'mac_address': self.mac1, 'type': 'physical',
-             'name': 'eth9',
-             'subnets': [{'type': 'dhcp4'}, {'type': 'dhcp6'}]}]}
+        expected = {'version': 2, 'ethernets': {'eth9': {
+            'match': {'macaddress': self.mac1.lower()}, 'set-name': 'eth9',
+            'dhcp4': True, 'dhcp6': True}}}
         self.assertEqual(
             expected,
             ec2.convert_ec2_metadata_network_config(
                 network_metadata_both, macs_to_nics))
 
+    def test_convert_ec2_metadata_network_config_handles_multiple_nics(self):
+        """When dhcp6 is public and dhcp4 is set to local enable both."""
```

Comment:
what does "dhcp4 is set to local"  mean?

Reply:
This is a poor docstring. I looked through our Ec2 network config and we aren't really doing anything with public addresses that are listed in metadata, maybe public addresses are something that we should be adding to the network config for Ec2? We should discuss and maybe add a followup that would add that support if warranted.

I think my initial docstring here was assuming on glance that we were actually using values provided by IMDS metadata for the 'public-ipv4s' metadata key to configure those static addresses on the interface. But as I found out (and dropped in this branch). We don't use that config for anything other that enabling DHCP4: true, which we also do if local-ipv4s or ipv6s have any values in them.

I *think* that covers all ec2 VMs, as any VM that has a publicly allocated VM ip will minimally have an ipv4 or ipv6 local ip as well in order to talk to IMDS.

Reply:
OK, let's clear that up

Reply:
set docstring to
"""DHCP route-metric increases on secondary NICs for IPv4 and IPv6"""